hasher-matcher-actioner/src/OpenMediaMatch/storage/postgres/database.py

```diff
@@ -456,6 +462,11 @@ def commit_signal_index(
         self.serialized_index_large_object_oid = l_obj.oid
         db.session.add(self)
         raw_conn.commit()
+        # Ensure raw connection is closed to avoid resource leaks
+        try:
+            raw_conn.close()
+        except Exception:
+            pass
```

Comment:
Hmm, my impression is that flask was handling this, and it was still in use.

If you want to make this less error prone, is there a place we can add `with` or the `closing` helper?